subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/TestDescriptorInternal.java

```diff
@@ -35,4 +35,5 @@ public interface TestDescriptorInternal extends TestDescriptor {
      */
     String getClassDisplayName();
 
+    boolean isParameterizedTest();
```

Comment:
❌ I don't think this method should be added. Instead, I think we should collect data for all non top-level suite descriptors.

Reply:
@marcphilipp Can you elaborate?  Why do you feel like checking for a non top-level suite would be better than explicitly modeling this in some way?  The fundamental problem here (as I understand it at least) is that our data model for recording test results doesn't account for test suites that don't also have associated tests with results.  So when a parameterized test is skipped, we miss it because there are no tests associated with it.  It feels wrong to me to assume that a suite that has a parent and is skipped _automatically_ doesn't have associated tests with results.  That's true for junit parameterized tests, but it doesn't seem right to assume that would be the case for all frameworks.  

It just seems like it's better to have the test framework tell us that a skipped suite won't have tests in some way, rather than inferring that after the fact.  This particular boolean could certainly be named in a less junit-centric fashion, but I'm not really sure what a better name is right now.  Stepping back, the model for recording results really should account for this problem in a more holistic way, but that's maybe a bigger effort than what this PR is trying to accomplish.

Reply:
IIUC we could simply collect data for all skipped descriptors regardless whether they are containers (a.k.a suites) or not. According to JUnit Platform semantics, a container that is skipped must not have any children that have been executed. The only way would be to report it as "aborted".

Reply:
I certainly like the simplicity of just doing this for any skipped container.  Would the junit semantics hold for TestNG and XCTest as well?

Reply:
Just poking around in the XCTest docs, it looks like there isn't an analog.  You can only skip tests from within a test method, and there is no way (that I could find at least) to skip a container of tests.  This is probably a better approach, then (just capturing skipped results for all descriptors, containers or not).

Reply:
@ghale I hope I understand the discussion correctly, I've updated this PR, removed the `isParameterizedTest` method. 

I also thought about adding more tests against TestNG, indeed the TestNG reports skip tests, but I didn't see this behavior in `Maven 3.8.1`, so I think maybe it's better to do that later, until it is discussed and considered needed.

Reply:
Sorry for the back and forth on this.  What Mark was pointing out is that we should be able to capture all "skipped" events, whether it is a suite or not.  We already capture skipped tests, and you've added something to capture skipped suites for non-top-level suites, but we should be able to just capture _all_ skipped suites.  There doesn't seem to be a situation where a skipped suite might have executed children, so this should be safe.